src/main/java/org/json/XML.java

```diff
@@ -445,6 +448,48 @@ private static boolean parse(XMLTokener x, JSONObject context, String name, XMLP
             }
         }
     }
+    /**
+     * This method removes any JSON entry which has the key set by XMLParserConfiguration.cDataTagName
+     * and contains whitespace as this is caused by whitespace between tags. See test XMLTest.testNestedWithWhitespaceTrimmingDisabled.
+     * @param jsonObject JSONObject which may require deletion
+     * @param config The XMLParserConfiguration which includes the cDataTagName
+     */
+    private static void removeEmpty(final JSONObject jsonObject, final XMLParserConfiguration config) {
+        if (jsonObject.has(config.getcDataTagName()))  {
+            final Object s = jsonObject.get(config.getcDataTagName());
+            if (s instanceof String) {
```

Comment:
Isn't CDATA content always a string?

Reply:
We are checking here if the value in the JSONObject at key CData is a string or JSONArray to remove it correctly if required.

Reply:
This check seems incorrect, I believe CDATA is always a string. Do you have a counterexample?

Reply:
CDATA itself is a string but the value at the tag labelled CDATA is possibly an array as we found in our testing due to some weird behaviour when whitespace in between closing and opening tags gets added to the result instead of trimmed. This is explained a bit more in my comment below